File: backend/src/main/java/be/lutske/leolegacy/interfaceadapter/rest/RecipeDetailResponse.java
package be.lutske.leolegacy.interfaceadapter.rest;

import java.util.List;

/**
 * JSON response DTO for a full recipe detail.
 */
public record RecipeDetailResponse(
        long id,
        String title,
        List<String> ingredients,
        String preparation,
        long categoryId,
        String categoryName,
        int viewCount
) {
}

File: backend/src/main/kotlin/be/lutske/leolegacy/interfaceadapter/rest/RecipeDetailResponse.kt
package be.lutske.leolegacy.interfaceadapter.rest

/**
 * JSON response DTO for a full recipe detail.
 */
data class RecipeDetailResponse(
    val id: Long,
    val title: String,
    val ingredients: List<String>,
    val preparation: String,
    val categoryId: Long,
    val categoryName: String,
    val viewCount: Int
)
